drivers/sensor/lps2xdf/lps2xdf.c

```diff
@@ -0,0 +1,491 @@
+	.avg = DT_INST_PROP(inst, avg),                                                            \
+	.drdy_pulsed = DT_INST_PROP(inst, drdy_pulsed),                                            \
+	.chip_api = &name##_chip_api,                                                              \
+	IF_ENABLED(DT_NODE_HAS_COMPAT(DT_DRV_INST(inst), st_lps28dfw),                             \
+		   (.fs = DT_INST_PROP(inst, fs),))                                                \
+	IF_ENABLED(DT_INST_NODE_HAS_PROP(inst, drdy_gpios), (LPS2XDF_CFG_IRQ(inst)))
+
+#define LPS2XDF_SPI_OPERATION (SPI_WORD_SET(8) | SPI_OP_MODE_MASTER | SPI_MODE_CPOL | SPI_MODE_CPHA)
+
+#define LPS2XDF_CONFIG_SPI(inst, name)                                                             \
+{                                                                                                  \
+	STMEMSC_CTX_SPI(&lps2xdf_config_##name##_##inst.stmemsc_cfg),                              \
+		.stmemsc_cfg =                                                                     \
+			{                                                                          \
+				.spi = SPI_DT_SPEC_INST_GET(inst, LPS2XDF_SPI_OPERATION, 0),       \
+			},                                                                         \
+		LPS2XDF_CONFIG_COMMON(inst, name)                                                  \
+}
+
+#define LPS2XDF_CONFIG_I2C(inst, name)                                                             \
+{                                                                                                  \
+	STMEMSC_CTX_I2C(&lps2xdf_config_##name##_##inst.stmemsc_cfg),                              \
+		.stmemsc_cfg =                                                                     \
+			{                                                                          \
+				.i2c = I2C_DT_SPEC_INST_GET(inst),                                 \
+			},                                                                         \
+		LPS2XDF_CONFIG_COMMON(inst, name)                                                  \
+}
+
+#define LPS2XDF_CONFIG_I3C(inst, name)                                                             \
+{                                                                                                  \
+	STMEMSC_CTX_I3C(&lps2xdf_config_##name##_##inst.stmemsc_cfg),                              \
+		.stmemsc_cfg =                                                                     \
+			{                                                                          \
+				.i3c = &lps2xdf_data_##name##_##inst.i3c_dev,                      \
+			},                                                                         \
+		.i3c.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),                                       \
+		.i3c.dev_id = I3C_DEVICE_ID_DT_INST(inst),                                         \
+			      LPS2XDF_CONFIG_COMMON(inst, name)                                    \
```

Comment:
should prob be something like

```c
        STMEMSC_CTX_I3C(&lps2xdf_config_##name##_##inst.stmemsc_cfg),                              \
        .stmemsc_cfg = {                                                                           \
                .i3c = &lps2xdf_data_##name##_##inst.i3c_dev,                                      \
        },                                                                                         \
        .i3c.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),                                               \
        .i3c.dev_id = I3C_DEVICE_ID_DT_INST(inst),                                                 \
        LPS2XDF_CONFIG_COMMON(inst, name) 
```

same for the two above, the current alignment is confusing and clang-format makes a mess as usual :-)